src/Enum.php

```diff
@@ -57,17 +57,36 @@ public function __construct(string $value = null)
         $this->value = $value;
     }
 
-    public static function __callStatic($name, $arguments): Enum
+    public static function __callStatic($name, $arguments)
     {
-        $enumValues = self::resolve();
+        if (strlen($name) > 2 && strpos($name, 'is') === 0) {
+            if (! isset($arguments[0])) {
+                throw new \ArgumentCountError(sprintf('Calling %s::%s() in static context requires one argument', static::class, $name));
+            }
+
+            return static::from($arguments[0])->$name();
+        }
 
-        if (! isset($enumValues[strtolower($name)])) {
+        if (! isset(self::resolve()[strtolower($name)])) {
             throw new TypeError("Method {$name} not available in enum ".static::class);
         }
 
         return new static($name);
     }
 
+    public function __call($name, $arguments)
+    {
+        if (strlen($name) > 2 && strpos($name, 'is') === 0) {
+            return $this->equals(substr($name, 2));
+        }
+
+        if (isset(self::resolve()[strtolower($name)])) {
```

Comment:
When exactly would this be used? Can you give an example?

Reply:
It cost me some time - the reason was the RecursiveEnum. The problem is that PHP doesn't call methods statically if they are called in `$this` context.

So the `__call()` method has to pass this to `__callStatic()`.

Reply:
https://stackoverflow.com/questions/13232445/call-catches-static-method-calls

Reply:
Right, and when would the static method be called in a `$this` context? It sounds to me like now someone is able to do this:

```php
$status = Status::draft();

$otherStatus = $status->published();
```

Did I understand it wrong?

Reply:
Yes he is - blame PHP. I've no idea who would need this but otherwise the tests with the `RecursiveEnum` class fail. Because it calls `static::foo()` in `$this->__toString()` which is handled by `__call()` instead of `__callStatic()` because a context is present.

Reply:
Oh well, this isn't the first strange PHP quirk I came across today. I guess it's fine ¯\_(ツ)_/¯